bondcpp/src/bond.cpp

```diff
@@ -437,21 +432,27 @@ bool Bond::waitUntilBroken(rclcpp::Duration timeout)
       break;  //  The deadline has expired
     }
     r.sleep();
+
+    std::unique_lock<std::mutex> lock(state_machine_mutex_);
+    if (sm_.getState().getId() == SM::Dead.getId()) {
+      broken = true;
+      break;
+    }
   }
 
-  return sm_.getState().getId() == SM::Dead.getId();
+  return broken;
 }
 
 bool Bond::isBroken()
 {
-  std::unique_lock<std::mutex> lock(mutex_);
+  std::unique_lock<std::mutex> lock(state_machine_mutex_);
   return sm_.getState().getId() == SM::Dead.getId();
 }
 
 void Bond::breakBond()
 {
   {
-    std::unique_lock<std::mutex> lock(mutex_);
+    std::unique_lock<std::mutex> lock(state_machine_mutex_);
     if (sm_.getState().getId() != SM::Dead.getId()) {
       sm_.Die();
       publishStatus(false);
```

Comment:
Can we drop the `state_machine_mutex` when calling `publishStatus`?

Reply:
Introduced some new private accessor functions for accessing the state machine state to scope the locking a little better, let me know what you think.